### PR TITLE
Module/users

### PR DESCRIPTION
**Pull request by mich71095**
### Changes ###
 - Setup "users" app
 - Create user model
 - User model testcase

### CODE REVIEW

File: users/tests/models_test.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+        user.delete()
+
+        with self.assertRaises(User.DoesNotExist):
+            User.objects.get(email=user_data['email'])
+
+    def test_create_user_fail(self):
+        """ Create user account with invalid data
+        """
+        new_user_data = user_data.copy()
```
Comment (earvingemenez) [inline users/tests/models_test.py:59]: good approach, good job

File: users/tests/models_test.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        new_user_data['first_name'] = None
```
Comment (earvingemenez) [inline users/tests/models_test.py]: use dict.update() instead of that

File: users/tests/models_test.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        self.assertIsInstance(user, User)
+        self.assertEqual(user.__str__(), user_data['email'])
+
+    def test_update_user(self):
+        """ Update the existing user and save
+        """
+        user = User.objects.create_user(**user_data)
+        self.assertIsInstance(user, User)
+
+        new_first_name = "Mark"
```
Comment (earvingemenez) [inline users/tests/models_test.py:38]: no need to assign the first_name value to a variable if you are not going to reuse it.

directly assign the value to the field.
`user.first_name = "Mark"`

File: users/tests/models_test.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        with self.assertRaises(IntegrityError):
+            User.objects.create_user(**new_user_data)
+
+    def test_update_user_fail(self):
+        """ Update the existing user with invalid data
+        """
+        user = User.objects.create_user(**user_data)
+        self.assertIsInstance(user, User)
+
+        new_first_name = None
```
Comment (earvingemenez) [inline users/tests/models_test.py]: do not assign a value to a variable if you are not going to re-use it

File: users/tests/models_test.py (hunk 3 above)
Comment (mich71095) [reply users/tests/models_test.py:38]: Sir, the variable new_first_name = "Mark", I reuse it from the last line of code in this method.     
self.assertEqual(user.first_name, new_first_name) which check the user first_name.

**Pull request merged**